src/clientiface.cpp

```diff
@@ -90,13 +91,26 @@ void RemoteClient::GetNextBlocks (
 		return;
 	}
 
+	// get view range and camera fov from the client
+	s16 wanted_range = sao->getWantedRange();
+	float camera_fov = sao->getFov();
+
+	const s16 full_d_max = std::min(g_settings->getS16("max_block_send_distance"), wanted_range);
+	const s16 d_opt = std::min(g_settings->getS16("block_send_optimize_distance"), wanted_range);
+	const s16 d_blocks_in_sight = full_d_max * BS * MAP_BLOCKSIZE;
+
 	v3f playerpos = sao->getBasePosition();
 	const v3f &playerspeed = player->getSpeed();
 	v3f playerspeeddir(0,0,0);
-	if(playerspeed.getLength() > 1.0*BS)
-		playerspeeddir = playerspeed / playerspeed.getLength();
-	// Predict to next block
-	v3f playerpos_predicted = playerpos + playerspeeddir*MAP_BLOCKSIZE*BS;
+	f32 playerspeedlen = playerspeed.getLength();
+	if (playerspeedlen > 1.0f * BS)
+		playerspeeddir = playerspeed / playerspeedlen;
+
+	// Predict where player will be soon, load blocks around there first
+	// We use the client-reported speed as an indicator, but do not move the
+	// center point further than 1/2 the visible range away
+	v3f playerpos_predicted = playerpos + playerspeeddir *
```

Comment:
How about optimising this by mving this calculation into the condition above so that it only occurs when player speed > 1 * BS?
So
```
v3f playerpos_predicted = playerpos;
if (playerspeedlen > 1.0f * BS) {
	v3f playerspeeddir = playerspeed / playerspeedlen;
	playerpos_predicted += playerspeeddir * rangelim(playerspeedlen * BS, MAP_BLOCKSIZE * BS, d_blocks_in_sight * 0.5f);
}
```
? and delete line 104.

Reply:
Optimizing is not needed here.
You mean for readability...? There I would agree.

Reply:
I know it's not hot code, but we still optimise code everywhere as these small optimisations do add up, over time this helps to mkae MT more lightweight. So in my opinion there is no such thing as 'optimising is not needed here'.
Also my suggestion just seems better and cleaner coding, avoiding doing calculations just to multiply those by 0 for stationary players.